src/Service.Tests/Unittests/SqlQueryExecutorUnitTests.cs

```diff
@@ -318,6 +318,63 @@ Mock<MsSqlQueryExecutor> queryExecutor
             Assert.AreEqual(3, queryExecutorLogger.Invocations.Count);
         }
 
+        /// <summary>
+        /// Validates streaming logic for QueryExecutor
+        /// In this test the DbDataReader.GetChars method is mocked to return 1024*1024 bytes (1MB) of data.
+        /// Max available size is set to 5 MB.
+        /// Based on number of loops, the data read will be 1MB * readDataLoops.Exception should be thrown in test cases where we go above 5MB.
+        /// This will be in cases where readDataLoops > 5.
+        /// </summary>
+        [DataTestMethod, TestCategory(TestCategory.MSSQL)]
+        [DataRow(4, false,
+            DisplayName = "Max available size is set to 5MB.Reading 1MB of data over 4 loops should succeed as amount of data being read is less than max-db-response-size-mb")]
+        [DataRow(5, false,
+            DisplayName = "Max available size is set to 5MB.Reading 1MB of data over 5 loops should succeed as amount of data being read is equal to max-db-response-size-mb")]
+        [DataRow(6, true,
+            DisplayName = "Max available size is set to 5MB.Reading 1MB of data over 6 loops should fail as amount of data being read exceeds max-db-response-size-mb")]
```

Comment:
Super nit for next future consideration. Easy to immediately identify what DataRow param means.
```suggestion
            DisplayName = "6 data read loop iterations * 1MB -> Fails to read 6MB because max-db-response-size-mb is 5MB")]
```

Reply:
fixed.